proxy/src/control/destination/mod.rs

```diff
@@ -228,10 +242,40 @@ impl Responder {
 
 // ===== impl Metadata =====
 
+impl Default for Metadata {
+    fn default() -> Self {
+        Self::no_metadata()
+    }
+}
+
 impl Metadata {
-    fn no_metadata() -> Self {
+    /// Construct a Metadata struct representing an endpoint with no metadata.
+    pub fn no_metadata() -> Self {
         Metadata {
-            metric_labels: None,
+            dst_labels: None,
+            // If we have no metadata on an endpoint, assume it does not support TLS.
+            tls_identity: None,
         }
     }
+
+    /// Construct a new Metadata with a set of labels from the Destination service.
+    pub fn new(
+        dst_labels: Option<DstLabels>,
+        tls_identity: Option<TlsIdentity>
+    ) -> Self {
+        Metadata {
+            dst_labels,
+            tls_identity,
+        }
+    }
+
+    /// Returns the endpoint's labels from the destination service, if it has them.
+    pub fn dst_labels(&self) -> Option<&DstLabels> {
+        self.dst_labels.as_ref()
+    }
+
+    /// Returns the endpoint's TLS verification strategy
```

Comment:
I think we should avoid using the term "verification strategy" since we really only have one strategy. This function returns the identity used for validating the endpoint's TLS certificate, which IMO is obvious from the name and signature.